modules/system/classes/asset/Mix.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace System\Classes\Asset;
+
+use Cms\Classes\Theme;
+use Illuminate\Support\HtmlString;
+use Winter\Storm\Support\Facades\Config;
+use Winter\Storm\Support\Facades\Url;
+use Winter\Storm\Support\Str;
+
+class Mix
+{
+    public static function mix(string $path, string $manifestDirectory = ''): HtmlString|string
+    {
+        static $manifests = [];
+
+        $theme = Theme::getActiveTheme();
```

Comment:
This should probably be in the CMS module instead since it requires the CMS module.

Reply:
@ericp-mrel any thoughts on the above?

Reply:
@jaxwilko do you have any input on the above?

Reply:
@ericp-mrel looked at your suggestion and I like it, I would leave the mix filter being registered by the CMS module however so that the default value of the package argument could be the current theme. Then if anyone wants to use it in plugins they can just use the Mix class directly (or if you see value in it add a addMix() method like we have addVite() in the AssetMaker trait).

Reply:
This is probably the approach I'd go with (in terms of replicating the `vite()` method functionality), however I'd probably try and keep the signatures the same (add support for array input with different file types) and return the entire tag i.e.

Twig:
```twig
{{ mix(['path/to/example.js', 'path/to/example.css'], 'theme-example') }}
```
Rendered output:
```html
<link rel="modulepreload" href="https://example.com/path/to/example-hash.js">
<script type="module" src="https://example.com/path/to/example-hash.js"></script>
<link rel="preload" as="style" href="https://example.com/path/to/example-hash.css">
<link rel="stylesheet" href="https://example.com/path/to/example-hash.css">
```

This will allow people to swap between vite and mix easier in the future and makes the documentation simplier as they will both function in the same way.

It's probably worth adding support for passing the manifest file as an extra param on the tag, as the file doesn't necessarily always have the same name (see: https://github.com/wintercms/winter/pull/1167) i.e.

```twig
{{ mix(['path/to/example.js', 'path/to/example.css'], 'theme-example') }}
// vs
{{ mix(['path/to/example.js', 'path/to/example.css'], 'theme-example', 'my-custom-manifest.json') }}
// or
{{ mix(['path/to/example.js', 'path/to/example.css'], 'theme-example', 'path/to/my-custom-manifest.json') }}
```

We can assume the default, but if a custom manifest name/path is passed it will allow for user overrides.